addons/stock/models/stock_move_line.py

```diff
@@ -268,9 +268,14 @@ def _apply_putaway_strategy(self):
                     smls.package_level_id.location_dest_id = smls.location_dest_id
             else:
                 for sml in smls:
-                    putaway_loc_id = sml.move_id.location_dest_id.with_context(exclude_sml_ids=excluded_smls)._get_putaway_strategy(
-                        sml.product_id, quantity=sml.quantity, packaging=sml.move_id.product_packaging_id,
-                    )
+                    if self.env.context.get('prefer_move_line_location'):
+                        putaway_loc_id = sml.location_dest_id.with_context(exclude_sml_ids=excluded_smls)._get_putaway_strategy(
+                            sml.product_id, quantity=sml.quantity, packaging=sml.move_id.product_packaging_id,
+                        )
+                    else:
+                        putaway_loc_id = sml.move_id.location_dest_id.with_context(exclude_sml_ids=excluded_smls)._get_putaway_strategy(
+                            sml.product_id, quantity=sml.quantity, packaging=sml.move_id.product_packaging_id,
+                        )
```

Comment:
it would be simpler to keep the common code in my opinion, eg:

```suggestion
                    if self.env.context.get('prefer_move_line_location'):
                        location_dest = sml.location_dest_id
                    else:
                        location_dest = sml.move_id.location_dest_id
                    putaway_loc_id = location_dest.with_context(exclude_sml_ids=excluded_smls)._get_putaway_strategy(
                        sml.product_id, quantity=sml.quantity, packaging=sml.move_id.product_packaging_id,
                    )                        
```
                        putaway_loc_id = sml.location_dest_id.with_context(exclude_sml_ids=excluded_smls)._get_putaway_strategy(
                            sml.product_id, quantity=sml.quantity, packaging=sml.move_id.product_packaging_id,
                        )
                    else:
                        putaway_loc_id = sml.move_id.location_dest_id.with_context(exclude_sml_ids=excluded_smls)._get_putaway_strategy(
                            sml.product_id, quantity=sml.quantity, packaging=sml.move_id.product_packaging_id,
                        )
```
also why I'm not sure why we can't check `qty_done` here (eventually still keeping the context key if it's necessary), but I don't know this code functionally at all

Reply:
`qty_done` is now a barcode-only field after the reworked quantity field update here: 7dda6bb92715